### PR TITLE
improve navigation highlighting:

### PR DESCRIPTION
**Pull request by salim-b**
- highlight blog section when on taxonomy sites (`/tags/*` and `/categories/*`)
- highlight home, too (same as PR #208)

The blog section is identified by its `identifier` key, which has the advantage that users can change the `name` and `url` of the blog menu item.

There is one basic assumption behind the code: There is only _one_ blog section in the whole site. I think this assumption is normally satisfied. If someone really wanted to create a site with multiple different blog sections, he/she would have to make other additional changes to this theme – which requires advanced knowledge of Hugo's internals anyway. For someone capable doing this, it should be rather trivial to also adjust the navigation partial to work as desired.

### CODE REVIEW

File: exampleSite/config.toml [hunk 1]
```diff
@@ -18,22 +18,25 @@ paginate = 10
...
 [[menu.main]]
+    identifier="blog"
     name = "Blog"
     url  = "/blog/"
     weight = 2
 
 [[menu.main]]
-    name = "FAQ"
     identifier = "faq"
+    name = "FAQ"
```
Comment (ryanfox1985) [inline exampleSite/config.toml]: ok, just alphabetical order.

**Comment by ryanfox1985** [comment]
I don't like to force to the people to set 'blog' to a their post/blog/docs pages.

- is it possible to change this condition? `eq .Identifier "blog"`
- I dislike have another parameter idea.

for instance: https://linode.com/docs

**Comment by salim-b** [comment]
> If at all possible, it would be good to have it set a default identifier for any menu items, possibly just a simple thing of it using the url with the slashes removed.
>
> `if not identifier theUrlStripped`

I don't understand what you mean... I've set the `identifier` key for all 4 menu items.

> I don't like to force to the people to set 'blog' to a their post/blog/docs pages.
> 
>     * is it possible to change this condition? `eq .Identifier "blog"`
> 
>     * I dislike have another parameter idea.
> 
> 
> for instance: https://linode.com/docs

I don't understand what you mean, either. We only "force" users to leave the `identifier` key untouched. They can still change the `url` and `name` as I've mentioned in the PR description.

**Comment by GeorgeWL** [comment]
>I don't understand what you mean, either. We only "force" users to leave the identifier key untouched. They can still change the url and name as I've mentioned in the PR description.

It's still a required field, cause if for example the user creates  new menu item, they must have a identifier param. 

It's adding unnecessary steps, and would be better if it didn't need to be set but could be overridden.

**Comment by salim-b** [comment]
> It's still a required field, cause if for example the user creates new menu item, they must have a identifier param.
> It's adding unnecessary steps, and would be better if it didn't need to be set but could be overridden.

Ok, now I get what you mean! You're right. I've added a fallback to `trim .URL "/"` if `.Identifier` is not set for a menu item. This makes setting the `identifier` key optional for new menu items.

**Comment by GeorgeWL** [comment]
Sounds good. I'd put an example into the exampleSite then I think it should be all good to pull right @ryanfox1985?

**Comment by salim-b** [comment]
> I'd put an example into the exampleSite

I'm afraid I don't understand what kind of example you mean. In the README I've added the following sentence:

> **Important:** Do not change the `identifier` key of existing menu entries!

What else is there to mention?

**Comment by GeorgeWL** [comment]
There's a folder called `examplesite` in the GitHub, that's shows a working example of a site.

**Comment by salim-b** [comment]
> There's a folder called `examplesite` in the GitHub, that's shows a working example of a site.

Yeah, I know. But what do you want to add there?

**Comment by GeorgeWL** [comment]
@salim-b an example.

**Comment by GeorgeWL** [comment]
> > There's a folder called `examplesite` in the GitHub, that's shows a working example of a site.
> 
> Yeah, I know. But what do you want to add there?

I missed that you'd already done it 🤦‍♂

**Pull request merged**